src/dwarf/Gparser.c

```diff
@@ -574,6 +584,8 @@ get_rs_cache (unw_addr_space_t as, intrmask_t *saved_maskp)
   if ((atomic_read (&as->cache_generation) != atomic_read (&cache->generation))
        || !cache->hash)
     {
+      /* cache_size is only set in the global_cache, copy it over before flushing */
+      cache->log_size = as->global_cache.log_size;
```

Comment:
We could punt for now and disallow set cache size for UNW_CACHE_PER_THREAD if you want.  Otherwise I think it requires a pthread key destructor.